### ISSUE TITLE
Split secrets API

### ISSUE THREAD

**Issue by Szewcson** [QubesOS/qubes-issues#7815]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)
When You want use multiple apps like chromium/chrome (I understand that it is only important if you plan to use builtin password manager), spotify (since it is chromium based?), Skype (which in fact using keyrings for storing session secrets) running in separate VMs You have to provide multiple gnome-keyring (or keepassxc if configured as secret service agent) passwords which is inconvenient. Alternatively you can disable storing master  password for chromium in the keyrings but this isn't a secure default since it then encrypting password manager using known default password. I mentioned it in https://github.com/QubesOS/qubes-issues/issues/5080 since the solution can be very similar.

### The solution you'd like
Some mechanism analogical to gpg/ssh split, that will allow storing secrets in one secret service agent in a separate VM.

### The value to a user, and who that user might be
It will simplify UX for not advanced users, and increase security by not storing secrets in potentially vulnerable app VMs.

**Comment by andrewdavidwong** [comment]
I don't understand. Are you suggesting some kind of "split password manager"? How is that different from just using a password manager in a vault qube and copy/pasting passwords from it into other app qubes? If the only difference is "autofill," wouldn't this be a duplicate of #5080?

Also, what is a "secret service API"?

**Comment by Szewcson** [comment]
Difference is in that #5080 is about passwords set by user (and browser integration) and here I writing about secret service API which are secrets generated by programs itself, or user passwords used in programs stored in the system. As I wrote there is possibility to use KeepasXC as secret service agent but normally in Debian systems this agent is gnome-keyring or Kwallet.

https://specifications.freedesktop.org/secret-service/latest/ch01.html

I try googling if gpg-agent (which will allow to use split-gpg) can be used as secret service agent but without a success.

**Comment by andrewdavidwong** [comment]
Can you provide some example use cases? I'm still struggling to see how this would be useful.

For example, Google Chrome uses gnome-keyring if I save some a website password using Chrome's built-in password-saving feature. But in such cases, I am intentionally choosing to save my password in Chrome in that app qube instead of saving it in my password manager in my offline vault qube and copy/pasting it to the online qube each time. So, I don't see how I'm missing any features or how what's being proposed in this issue would improve things in my example scenario.

Also, what is the difference between "passwords set by user" and "user passwords used in programs stored in the system"? Isn't the latter a proper subset of the former?

**Comment by Szewcson** [comment]
For chrome the use case is that:
We open chrome for the first time, and we are prompted to create new keyring password (for which in normal OS our user password will be used), this password is used to secure keyring in which chrome stores auto generated password which then is used to encrypt database of passwords stored in chrome built in password manager.

Then on every start of chrome we are prompted to provide that keyring password - which normally will be taken from our login screen. If we have few VMs with chrome we have to enter few passwords just to start chrome in that VMs.

If we decide to use week password or use option to not encrypt passwords stored in chrome database (which in fact use default password for encryption) our passwords can be simply brute forced. 

But if we instead keyring strong-password will be auto typed by qubes agent everything will work smooth and secure.

For Skype use case is that:
When we start skype for the first time we are prompted to create keyring password in the same manner as for chrome. Then we login to skype, and it create some session secrets for each user and stores it in that keyring.

On every skype start we are prompted to enter keyring password. Then skype is started and we are loged in automatically. 
So again if we can provide keyring password from some keys vm we have smooth and secure experience just like on normal OS where all the keyring magic is transparent for the user. Also this will be more secure then on normal OS because we can create very strong password that will be entered automatically from other vm and stored securely, guarded by keyfile, u2f and rationally strong password.

**Comment by Szewcson** [comment]
> Yes, if you want to save it in that app qube, then the feature is not for your case. But if you want to save it in a vault qube and have a bit more convenient usage (just qrexec confirmation, instead of copying the password manually), then secrets API could be useful. This may be also nice for browsers running in disposable qubes, where "local" password manager is rather useless (with split-"secrets API", you can have everything disposable, with exception to passwords).

Not exactly, since as I explained it before, that password is encrypted in db by key that is stored in the keyring, that is encrypted by password that I propose to pass automatically.
